### ISSUE TITLE
OpenAI integration is broken

### ISSUE THREAD

**Issue by guseggert** [approximatelabs/datadm#9]
When trying to use an OpenAI model, no response is returned and DataDM shows "Error" in the middle of the screen.

Logs:

```
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/gradio/routes.py", line 437, in run_predict
    output = await app.get_blocks().process_api(
  File "/usr/local/lib/python3.10/site-packages/gradio/blocks.py", line 1352, in process_api
    result = await self.call_function(
  File "/usr/local/lib/python3.10/site-packages/gradio/blocks.py", line 1093, in call_function
    prediction = await utils.async_iteration(iterator)
  File "/usr/local/lib/python3.10/site-packages/gradio/utils.py", line 341, in async_iteration
    return await iterator.__anext__()
  File "/usr/local/lib/python3.10/site-packages/gradio/utils.py", line 334, in __anext__
    return await anyio.to_thread.run_sync(
  File "/usr/local/lib/python3.10/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/usr/local/lib/python3.10/site-packages/gradio/utils.py", line 317, in run_sync_iterator_async
    return next(iterator)
  File "/datadm/datadm/app.py", line 49, in bot
    yield from agent.bot(repl, conversation, model_selection)
  File "/datadm/datadm/agent.py", line 22, in bot
    for conversation in self._bot(repl, conversation, llm):
  File "/datadm/datadm/agents/baseline.py", line 57, in _bot
    for result in precode(conversation=clean_conversation_list(starting_convo), silent=True, stream=True):
  File "/usr/local/lib/python3.10/site-packages/guidance/_program.py", line 251, in _stream_run
    loop.run_until_complete(execute_task) # this will stop each time the program wants to emit a new state
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/guidance/_program.py", line 385, in execute
    await self._executor.run(llm_session)
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 106, in run
    raise e
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 102, in run
    await self.visit(self.parse_tree)
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 437, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 437, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 404, in visit
    command_output = await command_function(*positional_args, **named_args)
  File "/usr/local/lib/python3.10/site-packages/guidance/library/_assistant.py", line 13, in assistant
    return await role(name="assistant", hidden=hidden, _parser_context=_parser_context)
  File "/usr/local/lib/python3.10/site-packages/guidance/library/_role.py", line 14, in role
    out = await parser.visit(
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 437, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 437, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 226, in visit
    visited_children = [await self.visit(child, next_node, next_next_node, prev_node, node, parent_node) for child in node.children]
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 226, in <listcomp>
    visited_children = [await self.visit(child, next_node, next_next_node, prev_node, node, parent_node) for child in node.children]
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 437, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/usr/local/lib/python3.10/site-packages/guidance/_program_executor.py", line 300, in visit
    command_output = await command_function(*positional_args, **named_args)
  File "/usr/local/lib/python3.10/site-packages/guidance/library/_gen.py", line 155, in gen
    for resp in gen_obj:
  File "/usr/local/lib/python3.10/site-packages/guidance/llms/_openai.py", line 192, in stream_then_save
    for curr_out in gen:
  File "/usr/local/lib/python3.10/site-packages/guidance/llms/_openai.py", line 44, in add_text_to_chat_mode_generator
    c['text'] = c['delta']['content']
  File "/usr/local/lib/python3.10/site-packages/openai/openai_object.py", line 71, in __setitem__
    raise ValueError(
ValueError: You cannot set text to an empty string. We interpret empty strings as None in requests.You may set {
  "index": 0,
  "delta": {
    "role": "assistant",
    "content": ""
  },
  "finish_reason": null
}.text = None to delete the property
```

Looks like this is due to: https://github.com/microsoft/guidance/issues/245

**Comment by guseggert** [comment]
fix https://github.com/approximatelabs/datadm/commit/de32688dba9b5b3debccbf92392990befe6a7cbf

Once Docker build completes I'll update the website to use it

**Comment by guseggert** [comment]
This is fixed in 0.3.5
